src/Core/Checkout/Customer/Validation/Constraint/CustomerPasswordMatches.php

```diff
@@ -29,31 +30,40 @@ class CustomerPasswordMatches extends Constraint
     /**
      * @param ?array{salesChannelContext: SalesChannelContext} $options
      *
-     * @deprecated tag:v6.8.0 - Parameter $options will be required and natively typed as array
+     * @deprecated tag:v6.8.0 - reason:new-optional-parameter - $options parameter will be removed, use $salesChannelContext instead
+     * @deprecated tag:v6.8.0 - reason:new-optional-parameter - $salesChannelContext parameter will required
      *
      * @internal
      */
-    public function __construct($options = null)
+    #[HasNamedArguments]
+    public function __construct(?array $options = null, ?SalesChannelContext $salesChannelContext = null)
     {
-        if ($options === null) {
-            Feature::triggerDeprecationOrThrow('v6.8.0.0', 'The parameter $options will be required and natively typed as array');
+        if ($options !== null || $salesChannelContext === null) {
+            Feature::triggerDeprecationOrThrow(
+                'v6.8.0.0',
+                Feature::deprecatedMethodMessage(self::class, __METHOD__, 'v6.8.0.0', 'Use $salesChannelContext argument instead of providing it in $options array')
+            );
         }
 
-        $options ??= [];
+        if ($options === null || Feature::isActive('v6.8.0.0')) {
+            if ($salesChannelContext === null) {
+                throw CustomerException::missingOption('salesChannelContext', self::class);
+            }
 
-        if (!Feature::isActive('v6.8.0.0') && isset($options['context'])) {
-            $options['salesChannelContext'] = $options['context'];
-        }
+            parent::__construct();
 
-        if (!($options['salesChannelContext'] ?? null) instanceof SalesChannelContext) {
-            throw CustomerException::missingOption('salesChannelContext', self::class);
-        }
+            $this->salesChannelContext = $salesChannelContext;
+        } else {
+            if (isset($options['context'])) {
+                $options['salesChannelContext'] = $options['context'];
+            }
 
-        if (!Feature::isActive('v6.8.0.0')) {
-            $options['context'] = $options['salesChannelContext'];
-        }
+            if (!($options['salesChannelContext'] ?? null) instanceof SalesChannelContext) {
+                throw CustomerException::missingOption('salesChannelContext', self::class);
+            }
 
-        parent::__construct($options);
+            parent::__construct($options);
+        }
     }
 
     /**
```

Comment:
same here, instead it could be deprecated to return the normal context :thinking: